src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -711,10 +712,14 @@ private Supplier<BigInteger> updateTotalCount(
             long total = getTotalCount(totalCountExecutors);
             return () -> BigInteger.valueOf(total);
         } else {
+            AtomicLong cache = new AtomicLong(Long.MIN_VALUE);
             return () -> {
                 try {
-                    long totalCount1 = getTotalCount(totalCountExecutors);
-                    return BigInteger.valueOf(totalCount1);
+                    if (cache.get() == Long.MIN_VALUE) {
+                        long totalCount1 = getTotalCount(totalCountExecutors);
+                        return BigInteger.valueOf(totalCount1);
```

Comment:
The cache should be updated here no?

Something  like this (untested):

```suggestion
                        long totalCount1 = getTotalCount(totalCountExecutors);
                        cache.set(totalCount1);
                        return BigInteger.valueOf(totalCount1);
```

Reply:
Yep, updated.